packages/client/lib/sync/fullsync.ts

```diff
@@ -69,17 +73,29 @@ export class FullSynchronizer extends Synchronizer {
       let oldHead = (await this.vm.blockchain.getHead()).hash()
       let newHead = Buffer.alloc(0)
       while (!newHead.equals(oldHead) && !this.stopSyncing) {
-        newHead = oldHead
+        oldHead = newHead
```

Comment:
@jochem-brouwer ah, and I also switched this since I was pretty much convinced after reading the code that this was swapped on first iteration. 😀  Let me know if I am wrong here.

Reply:
(so just the usage of `oldHead` and `newHead`)

Reply:
This would set `oldHead` immediately to the empty buffer, where `newHead` can never be (in this while loop) an empty Buffer, so this would only run one block. The names I have choosen might be a bit poorly choosen though.

The idea is to keep track of the current head in `blockchain`, run a block, and then check if the head is the same (in that case, no blocks were executed, so we can stop looping). If the head in the package changes, it means we have succesfully ran a block and we might succeed on running the next block as well. Note that `runBlockchain` calls the `iterator` of `blockchain`, which in turn updates the head in case blocks are being ran.

Reply:
Ah, I have just forgotten to switch the initialization lines (I just wanted to switch names, not any of the logic), let me fix.

Reply:
Fixed.